crates/task-impls/src/consensus.rs

```diff
@@ -219,13 +221,15 @@ async fn create_and_send_proposal<TYPES: NodeType>(
     upgrade_cert: Option<UpgradeCertificate<TYPES>>,
     proposal_cert: Option<ViewChangeEvidence<TYPES>>,
     round_start_delay: u64,
+    fee_data: FeeData<TYPES>,
```

Comment:
This parameter list is getting pretty painful. Is it possible to perhaps just build `CommitmentAndMetadata` and pass that in directly instead of adding yet-another parameter to this list? It looks like the caller will have access to everything.